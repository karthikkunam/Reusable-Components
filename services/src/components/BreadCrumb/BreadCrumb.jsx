import React from 'react';
import PropTypes from 'prop-types';
import {
  undefinedString,
} from './BreadCrumbConstants';

export default class BreadCrumb extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      routes: props.routes,
    };
  }

  static getDerivedStateFromProps(props, state) {
    if (props.routes !== state.routes) {
      return {
        routes: props.routes,
      };
    }
    // Return null if the state hasn't changed
    return null;
  }

  routeChange(href) {
    this.props.routeChangeFromBreadCrumb(href);
    // this.props.history.push(`${backSlashString}${href}`);
  }

  render() {
    const { routes } = this.state;
    return (
      <nav aria-label="breadcrumb">
        <ol className="breadcrumb">
          {
            routes && routes.length > 0 && routes.map((route, index) => {
              return (
                <li key={route.name} className={index === (routes.length - 1) ? 'active breadcrumb-item ' : 'breadcrumb-item'}>
                  {index !== (routes.length - 1) ?
                    <a key={route.name} onClick={() => { this.routeChange(route.href); }} aria-current="page">
                      {route && route.name ? route.name : undefinedString}
                    </a> :
                    route && route.name ? route.name : undefinedString
                  }
                </li>
              );
            })
          }
        </ol>
      </nav>
    );
  }
}

BreadCrumb.displayName = 'BreadCrumb';

BreadCrumb.propTypes = {
  routes: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.object])),
  routeChangeFromBreadCrumb: PropTypes.func
};

BreadCrumb.defaultProps = {
  routes: [{name:'Home', href:'https://www.google.com'}],
};
