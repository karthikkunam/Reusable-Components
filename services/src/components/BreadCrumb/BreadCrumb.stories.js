import React from 'react';
import { action } from '@storybook/addon-actions';
import BreadCrumb from '.';

export default {
  title: 'BreadCrumb',
  component: BreadCrumb,
};

 const route1 = [{name:'Home', href:'https://getbootstrap.com/docs/4.0/components/breadcrumb/#'}];
 const route2 = [{name:'Home', href:'https://getbootstrap.com/docs/4.0/components/breadcrumb/#'}, {name:'Products', href:'home/products'}];
 const route3 = [{name:'Home', href:'home'}, {name:'Products', href:'home/products'}, {name:'Library', href:'home/products/library'} ];

 export const BasicBreadCrumb = () => (
    <>
    <BreadCrumb routes={route1} routeChangeFromBreadCrumb={action('href')}/>
    <BreadCrumb routes={route2} routeChangeFromBreadCrumb={action('href')}/>
    <BreadCrumb routes={route3} routeChangeFromBreadCrumb={action('href')}/>
    </>
);
