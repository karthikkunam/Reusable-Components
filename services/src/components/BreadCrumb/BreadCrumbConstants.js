 
/* eslint-disable import/prefer-default-export */
/* istanbul ignore file */
export const basename = '/7boss/inventory/';
export const emptyString = '';
export const paramString = '?';
export const hashString = '#';
export const undefinedString = 'undefined';
export const backSlashString = '/';
export const pageRoutesToNames = {
  home: 'Inventory Management',
  recieving: 'Receiving',

};