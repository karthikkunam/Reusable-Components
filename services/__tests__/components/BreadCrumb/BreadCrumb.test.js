import React from 'react';
import { shallow } from 'enzyme';
import BreadCrumb from '../../../src/components/BreadCrumb';

const props = {
  routes: [
    { name: 'Home', href: 'https://getbootstrap.com/docs/4.0/components/breadcrumb/#' },
    { name: 'Products', href: 'home/products' },
  ],
  routeChangeFromBreadCrumb: jest.fn(),
};
describe('BreadCrumb Component', () => {
  it('renders correctly', () => {
    const component = shallow(<BreadCrumb {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('Should call routeChange function', () => {
    const component = shallow(<BreadCrumb {...props} />);
    component
      .find('ol')
      .at(0)
      .find('li')
      .at(0)
      .find('a')
      .simulate('click', {
        routeChange: () => {},
      });
    expect(component).toMatchSnapshot();
  });
});
