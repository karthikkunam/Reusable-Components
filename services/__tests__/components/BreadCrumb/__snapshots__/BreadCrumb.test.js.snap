// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumb Component Should call routeChange function 1`] = `
<nav
  aria-label="breadcrumb"
>
  <ol
    className="breadcrumb"
  >
    <li
      className="breadcrumb-item"
      key="Home"
    >
      <a
        aria-current="page"
        key="Home"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="active breadcrumb-item "
      key="Products"
    >
      Products
    </li>
  </ol>
</nav>
`;

exports[`BreadCrumb Component renders correctly 1`] = `
<nav
  aria-label="breadcrumb"
>
  <ol
    className="breadcrumb"
  >
    <li
      className="breadcrumb-item"
      key="Home"
    >
      <a
        aria-current="page"
        key="Home"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="active breadcrumb-item "
      key="Products"
    >
      Products
    </li>
  </ol>
</nav>
`;
